topics/request.md

```diff
@@ -1,316 +1,116 @@
 
-Complete `HeadersBuilder` API is listed [here](https://api.ktor.io/%ktor_version%/io.ktor.http/-headers-builder/).
-
-## Specifying a body for requests
+Learn how to run this sample from [client-upload-file](https://github.com/ktorio/ktor-documentation/tree/main/codeSnippets/snippets/client-upload-file).
 
-For `POST` and `PUT` requests, you can set the `body` property:
 
-```kotlin
-client.post<Unit> {
-    url("http://127.0.0.1:8080/")
-    body = // ...
-}
-```
 
-The `HttpRequestBuilder.body` property can be a subtype of `OutgoingContent` as well as a `String` instance:
 
-* `body = "HELLO WORLD!"`
-* `body = TextContent("HELLO WORLD!", ContentType.Text.Plain)`
-* `body = ByteArrayContent("HELLO WORLD!".toByteArray(Charsets.UTF_8))`
-* `body = LocalFileContent(File("build.gradle"))`
-* `body = JarFileContent(File("myjar.jar"), "test.txt", ContentType.fromFileExtension("txt").first())`
-* `body = URIFileContent("https://en.wikipedia.org/wiki/Main_Page")`
-
-If you install the [JsonFeature](json-feature.md), and set the content type to `application/json`
-you can use arbitrary instances as the `body`, and they will be serialized as JSON:
+## Parallel requests {id="parallel_requests"}
 
+When sending two requests at once, the client suspends the second request execution until the first is finished. If you need to perform several requests at once, you can use [launch](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/launch.html) or [async](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/async.html) functions. The code snippet below shows how to perform two requests asynchronously:
 ```kotlin
-data class HelloWorld(val hello: String)
-
-val client = HttpClient(Apache) {
-    install(JsonFeature) {
-        serializer = GsonSerializer {
-            // Configurable .GsonBuilder
-            serializeNulls()
-            disableHtmlEscaping()
-        }
-    }
-}
-
-client.post<Unit> {
-    url("http://127.0.0.1:8080/")
-    body = HelloWorld(hello = "world")
-}
 ```
+{src="snippets/client-parallel-requests/src/main/kotlin/com/example/Application.kt" lines="12,19-22"}
 
-Alternatively (using the integrated `JsonSerializer`):
+To see a full example, go to [client-parallel-requests](https://github.com/ktorio/ktor-documentation/tree/main/codeSnippets/snippets/client-parallel-requests).
 
-```kotlin
-val json = io.ktor.client.features.json.defaultSerializer()
-client.post<Unit>() {
-    url("http://127.0.0.1:8080/")
-    body = json.write(HelloWorld(hello = "world")) // Generates an OutgoingContent
-}
-```
 
-Or using Jackson (JVM only):
+## Cancel a request {id="cancel-request"}
 
+If you need to cancel a request, you can cancel a coroutine that runs this request. For instance, the [launch](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/launch.html) function returns a `Job` that can be used to cancel the running coroutine:
```

Comment:
To cancel a request, cancel the coroutine that makes the request.

Not sure if "For instance" is needed here in this context.

Reply:
ok, removed